src/models/predicate/proven_v01.rs

```diff
@@ -0,0 +1,290 @@
+        let form: &str = Deserialize::deserialize(de)?;
+        DateTime::parse_from_rfc3339(form)
+            .map(TimeStamp)
+            .map_err(|e| DeserializeError::custom(format!("{:?}", e)))
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct TypeURI(pub(crate) String);
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct Builder {
+    pub(crate) id: TypeURI,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct Completeness {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) arguments: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) environment: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) materials: Option<bool>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct Recipe {
+    #[serde(rename = "type")]
+    pub(crate) typ: TypeURI,
+    #[serde(rename = "definedInMaterial")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) defined_in_material: Option<usize>,
+    #[serde(rename = "entryPoint")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) entry_point: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) arguments: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) environment: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Metadata {
+    #[serde(rename = "buildInvocationId")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) build_invocation_id: Option<String>,
+    #[serde(rename = "buildStartedOn")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) build_started_on: Option<TimeStamp>,
+    #[serde(rename = "buildFinishedOn")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) build_finished_on: Option<TimeStamp>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) completeness: Option<Completeness>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) reproducible: Option<bool>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Material {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) uri: Option<TypeURI>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) digest: Option<HashMap<String, String>>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+/// Predicate `ProvenV01` means the predicate of SLSA format.
+///
+/// [ProvenV01](https://slsa.dev/provenance/v0.1)
+/// can be used together with most states.
+pub struct ProvenV01 {
```

Comment:
I'd like this naming to improve. Here and in v02. Let's explicitly refer to it as SLSA Provenance so there is no confusion.